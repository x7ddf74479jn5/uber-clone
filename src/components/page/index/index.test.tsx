import { fireEvent, render, screen } from "@testing-library/react";

import { Home } from ".";

describe("Home page", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<Home />, {});
    expect(asFragment()).toMatchSnapshot();
  });

  it("clicking button triggers alert", () => {
    render(<Home />, {});
    window.alert = jest.fn();
    fireEvent.click(screen.getByText("Button"));
    expect(window.alert).toHaveBeenCalledWith("Hello, World!");
  });
});
