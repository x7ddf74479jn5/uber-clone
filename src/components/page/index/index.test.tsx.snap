// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <h2>
    Home
  </h2>
  <button>
    Button
  </button>
</DocumentFragment>
`;
